### PR TITLE
[core] Add background location permission

### PR DESCRIPTION
**Pull request by nagineni**
Add background location permission support in the PermissionsManager. Also, request location permissions based on the declarations in the manifest, instead of requesting fine location permission by default.

### CODE REVIEW

**Comment by tobrun** [comment]
@nagineni what is left to merge this PR? Would be cool if we could dogfood this downstream

**Review by mr1sunshine — APPROVED** [review]

**Comment by nagineni** [comment]
> @nagineni what is left to merge this PR? Would be cool if we could dogfood this downstream

I was waiting for the :heavy_check_mark: We are good to merge this now.

**Pull request merged**
